dcicutils/env_utils.py

```diff
@@ -369,23 +470,67 @@ def infer_repo_from_env(envname):
         return None
 
 
+# TODO: Figure out if these two actually designate different hosts or if we could make CGAP prettier. -kmp 4-Oct-2021
+CGAP_FOURSIGHT_URL_PREFIX = "https://u9feld4va7.execute-api.us-east-1.amazonaws.com/api/view/"
+FF_FOURSIGHT_URL_PREFIX = "https://foursight.4dnucleome.org/view/"
+
+
+def infer_foursight_url_from_env(request, envname):
+    token = infer_foursight_from_env(request, envname)
+    if token is not None:
+        prefix = CGAP_FOURSIGHT_URL_PREFIX if is_cgap_env(envname) else FF_FOURSIGHT_URL_PREFIX
+        return prefix + token
+    else:
+        return None
+
+
 def infer_foursight_from_env(request, envname):
     """  Infers the Foursight environment to view based on the given envname and request context
 
     :param request: the current request (or an object that has member 'domain')
     :param envname: name of the environment we are on
     :return: Foursight env at the end of the url ie: for fourfront-green, could be either 'data' or 'staging'
     """
-    if is_cgap_env(envname):
-        return envname[len('fourfront-'):]  # all cgap envs are prefixed and the FS envs directly match
+    if not envname or (not is_stg_or_prd_env(envname) and not envname.startswith('fourfront-')):
+        return None
+    elif is_cgap_env(envname):
+        return short_env_name(envname)  # all cgap envs are prefixed and the FS envs directly match
     else:
         if is_stg_or_prd_env(envname):
             if FF_PUBLIC_DOMAIN_PRD in request.domain:
                 return FF_PRODUCTION_IDENTIFIER
             else:
                 return FF_STAGING_IDENTIFIER
         else:
-            return envname[len('fourfront-'):]  # if not data/staging, behaves exactly like CGAP
+            return short_env_name(envname)  # if not data/staging, behaves exactly like CGAP
+
+
+def short_env_name(envname):
+    """
+    Given a short or long environment name, return the long name (e.g., with a 'fourfront-' prefix if not present).
```

Comment:
This is a misimpression. These are the legacy implementations, but when we bring in the other PR, it will handle `cgap-msa` or else none of this would be worth doing. This is part 1 of 2 that effects that change, and this code only needs to handle the legacy part.